Sources/LanguageServerProtocol/WorkspaceEdit.swift

```diff
@@ -11,12 +11,73 @@
 //===----------------------------------------------------------------------===//
 
 /// A workspace edit represents changes to many resources managed in the workspace.
-public struct WorkspaceEdit: Codable, Hashable, ResponseType {
+public struct WorkspaceEdit: Hashable, ResponseType, LSPAnyCodable {
 
   /// The edits to be applied to existing resources.
   public var changes: [URL: [TextEdit]]?
 
   public init(changes: [URL: [TextEdit]]?) {
     self.changes = changes
   }
+
+  public init?(fromLSPDictionary dictionary: [String : LSPAny]) {
+    guard case .dictionary(let lspDict) = dictionary[CodingKeys.changes.stringValue] else {
+      return nil
+    }
+    var dictionary = [URL: [TextEdit]]()
+    for (key, value) in lspDict {
+      guard let url = URL(string: key) else {
+        return nil
+      }
+      guard case .array(let array) = value else {
+        return nil
+      }
+      var edits = [TextEdit]()
+      for case .dictionary(let editDict) in array {
+        guard let edit = TextEdit(fromLSPDictionary: editDict) else {
+          return nil
+        }
+        edits.append(edit)
+      }
+      dictionary[url] = edits
+    }
+    self.changes = dictionary
+  }
+
+  public func encodeToLSPAny() -> LSPAny {
+    guard let changes = changes else {
+      return nil
+    }
+    let values = changes.map {
+      ($0.key.absoluteString, LSPAny.array($0.value.map { $0.encodeToLSPAny() }))
+    }
+    let dictionary = Dictionary(uniqueKeysWithValues: values)
+    return .dictionary([
+      CodingKeys.changes.stringValue: .dictionary(dictionary)
+    ])
+  }
+}
+
+// Workaround for Codable not correctly encoding dictionaries whose keys aren't strings.
+extension WorkspaceEdit: Codable {
+  enum CodingKeys: String, CodingKey {
+    case changes
+  }
+
+  public init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    self.changes = try container.decode([URL: [TextEdit]].self, forKey: .changes)
```

Comment:
It worked since URL encodes to a string, but it actually failed the `checkCoding` method for a similar reason as before so for safety I've now made it decode to the actual [String: [TextEdit]] type and proceed from there.